### PR TITLE
Fix APSP

### PR DESCRIPTION
**Pull request by angriman**
In disconnected graphs `APSP` returns distances that are less than infinity for pairs of nodes that are not reachable. This happens because`APSP` calls multiple times `BFS/Dijkstra::run()`, but neither `BFS` nor `Dijkstra` reset `distance` before running.
This PR also includes new tests that failed with the previous implementation.

### CODE REVIEW

File: networkit/cpp/distance/BFS.cpp [hunk 1]
```diff
@@ -20,12 +20,15 @@ void BFS::run() {
     reachedNodes = 1;
     sumDist = 0.;
 
+    constexpr auto infDist = std::numeric_limits<edgeweight>::max();
```
Comment (avdgrinten) [inline networkit/cpp/distance/BFS.cpp:23]: `constexpr` does nothing here, use `const` (or no qualifier).

File: networkit/cpp/distance/BFS.cpp [hunk 2]
```diff
@@ -20,12 +20,15 @@ void BFS::run() {
...
+    std::fill(distances.begin(), distances.end(), infDist);
+
     if (distances.size() < z) {
-        distances.resize(z, std::numeric_limits<edgeweight>::max());
+        distances.resize(z, infDist);
         visited.resize(z, ts);
     }
 
-    if (ts++ == 255) {
+    if (ts++ == std::numeric_limits<uint8_t>::max()) {
```
Comment (avdgrinten) [inline networkit/cpp/distance/BFS.cpp:31]: This change also does nothing (I am indifferent about it).

File: networkit/cpp/distance/BFS.cpp (hunk 1 above)
Comment (angriman) [reply networkit/cpp/distance/BFS.cpp]: Thanks, I using `const`.

File: networkit/cpp/distance/BFS.cpp (hunk 2 above)
Comment (angriman) [reply networkit/cpp/distance/BFS.cpp:31]: I think it's just better than having a magic number.

**Comment by angriman** [comment]
Since we are now resetting the `distance` vector at every run, does it make sense to keep using `visited`? The only reason to do so might be that `visited` uses 8-bit integers, while `distance` 64-bit floats, but we need benchmark results.

**Pull request merged**